app/main.py

```diff
@@ -2,24 +1,70 @@
+from typing import Any
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: object, name: str) -> None:
+        self.public_name = name
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: object, owner: object) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object, value: Any) -> None:
+        if not isinstance(value, int):
```

Comment:
This is a code quality issue. The `isinstance` function checks if `value` is an instance of `int`. However, it will also return `True` if `value` is an instance of a subclass of `int`. If you want to check if `value` is exactly an `int` (and not a subclass), you should compare `type(value)` with `int`.